syntax = "proto3";

package iroha.protocol;

import "block.proto";
import "google/protobuf/timestamp.proto";

enum ResponseCode {
  FAIL = 0; // default value
  OK = 1;
}

message ApiResponse {
  ResponseCode code = 1;
}

message PublishedBlock {
  Block block = 1;
  google.protobuf.Timestamp confirmedAt = 2;
}

message ToriiResponse {
  ResponseCode code = 1;
  string message = 2;
}

message VerifyResponse {
  ResponseCode code = 1;
  string message = 2;
}

message QueryResponse {}

message QueueTransactionResponse {
  ResponseCode code = 1;
  string message = 2;
}

service CommandService {
  rpc Torii (Transaction) returns (ToriiResponse);
}

service QueryService {
//  rpc Find (QueryRequest) returns (QueryResponse);
}

service SumeragiService {
  rpc Verify (Block) returns (VerifyResponse);
}

service OrderingService {
  rpc QueueTransaction (Transaction) returns (QueueTransactionResponse);
}

service SubscriberService {
  rpc receiveBlock (PublishedBlock) returns (ApiResponse);
}
